qt/place_page_dialog_user.cpp

```diff
@@ -0,0 +1,291 @@
+    }
+
+    // Operator fragment
+    if (auto operatorName = info.GetMetadata(feature::Metadata::EType::FMD_OPERATOR); !operatorName.empty())
+      addEntry("Operator", std::string(operatorName));
+
+    // Wifi fragment
+    if (info.HasWifi()){
+      addEntry("Wi-Fi", "Yes");
+    }
+
+    // Links fragment
+    if (auto website = info.GetMetadata(feature::Metadata::EType::FMD_WEBSITE); !website.empty())
+      addEntry("Website", stripSchemeFromURI(std::string(website)), true);
+
+    if (auto email = info.GetMetadata(feature::Metadata::EType::FMD_EMAIL); !email.empty()){
+      data->addWidget(new QLabel("Email"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='mailto:" + std::string(email) + "'>" + std::string(email) + "</a>"));
+      value->setOpenExternalLinks(true);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    if (auto facebook = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_FACEBOOK); !facebook.empty()){
+      data->addWidget(new QLabel("Facebook"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='" + osm::socialContactToURL(feature::Metadata::EType::FMD_CONTACT_FACEBOOK, std::string(facebook)) + "'>" + std::string(facebook) + "</a>"));
+      value->setOpenExternalLinks(true);
+      value->setTextInteractionFlags(Qt::TextBrowserInteraction);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    if (auto instagram = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_INSTAGRAM); !instagram.empty()){
+      data->addWidget(new QLabel("Instagram"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='" + osm::socialContactToURL(feature::Metadata::EType::FMD_CONTACT_INSTAGRAM, std::string(instagram)) + "'>" + std::string(instagram) + "</a>"));
+      value->setOpenExternalLinks(true);
+      value->setTextInteractionFlags(Qt::TextBrowserInteraction);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    if (auto twitter = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_TWITTER); !twitter.empty()){
+      data->addWidget(new QLabel("Twitter"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='" + osm::socialContactToURL(feature::Metadata::EType::FMD_CONTACT_TWITTER, std::string(twitter)) + "'>" + std::string(twitter) + "</a>"));
+      value->setOpenExternalLinks(true);
+      value->setTextInteractionFlags(Qt::TextBrowserInteraction);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    if (auto vk = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_VK); !vk.empty()){
+      data->addWidget(new QLabel("VK"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='" + osm::socialContactToURL(feature::Metadata::EType::FMD_CONTACT_VK, std::string(vk)) + "'>" + std::string(vk) + "</a>"));
+      value->setOpenExternalLinks(true);
+      value->setTextInteractionFlags(Qt::TextBrowserInteraction);
+
+      data->addWidget(value, row++, 1);
+    }
```

Comment:
Is it possible to avoid duplication for social networks by creating a lambda that takes a pair of parameters: "VK" and FMD_CONTACT_VK ? Can `addEntry` lambda be reused?

Reply:
I have created a new lambda for this.